Documentation: Docs changes for blog theme 2.0 (#25075)
gatsbyjs/gatsby: docs/docs/theme-ui.md

### `gatsby-plugin-theme-ui` configuration

`gatsby-plugin-theme-ui` allows for optional configuration. One of the options you can pass in is `preset`. Preset is a base theme object you'd like your site to use.

The `preset` can be a JSON object or a string package name that the plugin will require for you.

If you're passing in a string package name you need to make sure that the dependency is installed.

```shell
npm install @theme-ui/preset-funk
```

```js:title=gatsby-config.js
module.exports = {
  plugins: [
    {
      resolve: `gatsby-plugin-theme-ui`,
      options: {
        preset: "@theme-ui/preset-funk",
      },
    },
  ],
}
```

Alternatively, you could do the same thing and require the package yourself.

```shell
npm install @theme-ui/preset-funk
```

```js:title=gatsby-config.js
module.exports = {
  plugins: [
    {
      resolve: `gatsby-plugin-theme-ui`,
      options: {
        preset: require("@theme-ui/preset-funk"),
      },
    },
  ],
}
```

`preset` operates as your base theme and any local shadowed styles with automatically merge with it. Local styles are considered more specific and will override base styles where necessary.

### Local shadowing

> The Theme UI plugin uses the [component shadowing API][] to add the theme object context to your site.

Reviewer: The link at the bottom of this doc is wrong, it should be: https://www.gatsbyjs.org/docs/theme-api/#shadowing

I also think it would help to quickly define shadowing in case people are super new to Themes.

Theme UI uses a `theme` configuration object to provide color, typography, layout, and other shared stylistic values through [React context][].
This allows components within your site to add styles based on a predefined set of values.

There are two ways to add a theme to your site: using configuration options in `gatsby-plugin-theme-ui`, or local shadowing.

### `gatsby-plugin-theme-ui` configuration

`gatsby-plugin-theme-ui` allows for optional configuration. One of the options you can pass in is `preset`. Preset is a base theme object you'd like your site to use.

The `preset` can be a JSON object or a string package name that the plugin will require for you.

If you're passing in a string package name you need to make sure that the dependency is installed.

```shell
npm install @theme-ui/preset-funk
```

```js:title=gatsby-config.js
module.exports = {
  plugins: [
    {
      resolve: `gatsby-plugin-theme-ui`,
      options: {
        preset: "@theme-ui/preset-funk",
      },
    },
  ],
}
```

Alternatively, you could do the same thing and require the package yourself. This is necessary if you're referencing a local package that isn't available publicly.

```shell

> npm install @theme-ui/preset-funk

Reviewer 2: In this scenario, we've given the example of a local package that isn't available publicly. Could we carry that through to the code snippet (could be a made up name) rather than using the same package as in the previous example? I think that might help avoid some confusion.

```

```js:title=gatsby-config.js
module.exports = {
  plugins: [
    {
      resolve: `gatsby-plugin-theme-ui`,
      options: {
        preset: require("@theme-ui/preset-funk"),
      },
    },
  ],
}
```

> `preset` operates as your base theme and any local shadowed styles with automatically merge with it. Local styles are considered more specific and will override base styles where necessary.

Reviewer 2: ```suggestion
`preset` operates as your base theme and any local shadowed styles will automatically merge with it. Local styles are considered more specific and will override base styles where necessary.
```

I'm guessing at what this was supposed to say. 😅

## Creating a theme object

> Create a `src/gatsby-plugin-theme-ui` directory in your project, and add an `index.js` file to export a theme object. Add a `colors` object to the file to store the color palette for your site.

Reviewer 2: ```suggestion
Create a `src/gatsby-plugin-theme-ui` directory in your project and add an `index.js` file to export a theme object. Add a `colors` object to the file to store the color palette for your site.
```

## Using Theme UI in a Gatsby theme

When using Theme UI in a Gatsby theme, it's important to understand how the `gatsby-plugin-theme-ui` package handles theme objects from multiple Gatsby themes and sites.

If multiple themes are installed in the same site, the one defined last in your `gatsby-config.js` file's `plugins` array will take precedence.

> To extend an existing Theme UI configuration from a theme it's necessary to know how the base theme is passing styles. If it's using a `preset`, as the `gatsby-theme-blog`, any shadowed files are merged automatically.

Reviewer: Are there any tips you can add to help users make the leap to "know[ing] how the base theme is passing styles"? With this and the first sentence, the section seems a bit assumptive of intricate theme knowledge. Anything we can do to make it easier to understand would help. Here's an idea, feel free to edit for accuracy:
```suggestion
To extend an existing Theme UI configuration from a theme you'll need to find out how the base theme is passing styles. If it's using a `preset`, as the `gatsby-theme-blog`, any shadowed files are merged automatically. This can be discovered by looking at the theme source code in `node_modules` or on GitHub.
```

```js:title=src/gatsby-plugin-theme-ui/index.js
export default {
  colors: {
    text: "#222",
    primary: "tomato",
  },
}
```

> If the theme is shadowing styles itself, you'll want to merge those styles with your local ones.

Reviewer: ```suggestion
If the theme is shadowing styles itself without a preset, you'll want to merge those styles with your local ones. Here's an example using Theme UI's `merge` API:
```

## Using Theme UI in a Gatsby theme

When using Theme UI in a Gatsby theme, it's important to understand how the `gatsby-plugin-theme-ui` package handles theme objects from multiple Gatsby themes and sites.

If multiple themes are installed in the same site, the one defined last in your `gatsby-config.js` file's `plugins` array will take precedence.

> To extend an existing Theme UI configuration from a theme you'll need to find out how the base theme is passing styles. If it's using a `preset`, as the `gatsby-theme-blog`, any shadowed files are merged automatically. This can be discovered by looking at the theme source code in `node_modules` or on GitHub.

Reviewer 2: ```suggestion
To extend an existing Theme UI configuration from a theme you'll need to find out how the base theme is passing styles. If it's using a `preset`, as the `gatsby-theme-blog` package does, any shadowed files are merged automatically. This can be discovered by looking at the theme source code in `node_modules` or on GitHub.
```

```js:title=src/gatsby-plugin-theme-ui/index.js
export default {
  colors: {
    text: "#222",
    primary: "tomato",
  },
}
```

> If the theme is shadowing styles itself without a preset, you'll want to merge those styles with your local ones. Here's an example using Theme UI's `merge` API:

Reviewer 2: ```suggestion
If the theme itself is shadowing styles without a preset, you'll want to merge those styles with your local ones. Here's an example using Theme UI's `merge` API:
```
